src/Command/ValidateCommand.php

```diff
@@ -82,8 +91,62 @@ protected function doExecute(DependencyGraph $graph): int
             return 0;
         }
 
+        if ($this->input->getOption('generate-baseline')) {
+            $this->writeBaseline($graph, $violations);
+            $this->io->success(sprintf('Generated baseline containing %d ignored violations.', $violations->count()));
+        }
+
         $this->io->writeln('Try running the validator with the --autofix option to automatically fix most of the violations.');
 
         return 1;
     }
+
+    private function filterViolations(DependencyGraph $graph, ViolationCollection $violations): ViolationCollection
+    {
+        if (!file_exists($this->getBaselinePath($graph))) {
+            return $violations;
+        }
+
+        $contents = file_get_contents($this->getBaselinePath($graph));
+        if (false === $contents) {
+            throw new UnexpectedValueException('Could not read baseline.');
+        }
+
+        $ignored = Yaml::parse(file_get_contents($this->getBaselinePath($graph)) ?? '')['ignore'] ?? [];
+        if (!is_array($ignored)) {
+            throw new UnexpectedValueException('Expected ignored files to be array.');
+        }
+
+        return $violations->filter(static function (ViolationInterface $violation) use ($ignored): bool {
+            return !in_array($violation->getMessage(), $ignored, true);
+        });
+    }
+
+    private function writeBaseline(DependencyGraph $graph, ViolationCollection $violations): void
+    {
+        file_put_contents(
+            $this->getBaselinePath($graph),
+            Yaml::dump(['ignore' => $violations->getMessages()->toArray()])
+        );
+    }
+
+    private function getRules(InputInterface $input): RuleCollection
+    {
+        $rules = new RuleCollection([
+            new Rule\MissingDependencyRule(),
+            new Rule\IncompatibleVersionRule(),
+            new Rule\MissingReplaceRule(),
+        ]);
+
+        if ($input->getOption('static-analysis')) {
+            $rules->append(new Rule\OutOfBoundsClassReferenceRule($this->logger));
+        }
+
+        return $rules;
+    }
+
+    private function getBaselinePath(DependencyGraph $graph): string
+    {
+        return $graph->getRootPath() . DIRECTORY_SEPARATOR . 'composer-monolith-baseline.yaml';
```

Comment:
Would suggest to make a public constant for this file name, could be useful in the future to reference to it.